File: 프로그래머스/3/43162. 네트워크/네트워크.py
def dfs(current):
    global visited, answer
    stack = []
    stack.append(current)
    while stack:
        now = stack.pop()
        if visited[now]:
            continue
        visited[now] = 1
        for next in range(n):
            if now != next and computers[now][next]:
                stack.append(next)
    return

def solution(n, computers):
    answer = 0
    visited = [0] * n
    
    for i in range(n):
        if visited[i] == 0:
            answer += 1
            stack = []
            stack.append(i)
            while stack:
                now = stack.pop()
                if visited[now]:
                    continue
                visited[now] = 1
                for next in range(n):
                    if now != next and computers[now][next]:
                        stack.append(next)
            
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    var answer = 0;
    
    let visited = new Array(n).fill(0)
    
    // console.log(visited)
    
    for (i=0; i<n; i++) {
        if (visited[i]) continue;
        
        answer += 1;
        let stack = [i];
        while (stack.length) {
            let now = stack.pop();
            visited[now] = 1
            
            for (j=0; j<n; j++) {
                if (now != j && computers[now][j] && !visited[j]) {
                    stack.push(j)
                }
            }
        }
    }
    
    return answer;
}